File: concept/14_string/remove_substring_occurrance/remove_substring_occurrance.dart
void main() {
  // Leet Code 1910
  String s = "daabcbaabcbc";
  String part = "abc";

  print(s);
  while (s.length != 0 && s.contains(part)) {
    s = s.replaceAll(part, '');
    print(s);
  }

  print(s);
  /*
    i/o : daabcbaabcbc
    o/p : dab
    */
}

File: concept/14_string/remove_substring_occurrance/remove_substring_occurrance.cpp
#include <iostream>
using namespace std;

int main()
{
    // Leet Code 1910
    string s = "daabcbaabcbc";
    string part = "abc";

    cout << s << endl;
    while (s.length() != 0 && s.find(part) < s.length())
    {
        s.erase(s.find(part), part.length());
    }

    cout << s << endl;
    /*
    i/o : daabcbaabcbc
    o/p : dab
    */
}
